### PR TITLE
feature: add single send request for client.

### PR DESCRIPTION
**Pull request by objcoding**
### Ⅰ. Describe what this PR did
add single send request for client.
add 'transport.enable-client-batch-send-request' config param.

### CODE REVIEW

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientConfig.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
     private static final boolean DEFAULT_POOL_TEST_BORROW = true;
     private static final boolean DEFAULT_POOL_TEST_RETURN = true;
     private static final boolean DEFAULT_POOL_LIFO = true;
+    private static final boolean ENABLE_CLIENT_BATCH_SEND_REQUEST = CONFIG.getBoolean("transport.enable-client-batch-send-request", true);
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/rpc/netty/NettyClientConfig.java]: use constants

**Comment by objcoding** [comment]
> add configuration item in springboot starter and script

already added

**Comment by objcoding** [comment]
> add configuration item in springboot starter and script

**Comment by objcoding** [comment]
> add configuration item in springboot starter and script

already added

File: core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java [hunk 2]
```diff
@@ -230,24 +232,50 @@ private Object sendAsyncRequest(String address, Channel channel, Object msg, lon
...
+                        mergeLock.notifyAll();
+                    }
+                }
+            } else {
+                // the single send.
+                MergedWarpMessage mergeMessage = new MergedWarpMessage();
+                mergeMessage.msgs.add((AbstractMessage) rpcMessage.getBody());
+                mergeMessage.msgIds.add(rpcMessage.getId());
+                try {
+                    sendRequest(channel, mergeMessage);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java]: It ’s not the same as what I understand. The single piece I understand is to send RpcMessage directly.

File: core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java [hunk 3]
```diff
@@ -230,43 +230,42 @@ private Object sendAsyncRequest(String address, Channel channel, Object msg, lon
...
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("offer message: {}", rpcMessage.getBody());
+                }
+                if (!isSending) {
+                    synchronized (mergeLock) {
+                        mergeLock.notifyAll();
                     }
                 }
-            });
+            } else {
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java]: merge with next `else`

File: core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java (hunk 2 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java]: > It ’s not the same as what I understand. The single piece I understand is to send RpcMessage directly.

ok.

File: core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java (hunk 3 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemoting.java]: > merge with next `else`

ok.

File: core/src/main/java/io/seata/core/rpc/netty/NettyClientConfig.java (hunk 1 above)
Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/NettyClientConfig.java]: > use constants

I will create a new pr to change the netty part constant

File: core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemotingClient.java [hunk 4]
```diff
@@ -113,19 +113,23 @@ public void run() {
...
             }
         }, SCHEDULE_INTERVAL_MILLS, SCHEDULE_INTERVAL_MILLS, TimeUnit.SECONDS);
-        mergeSendExecutorService = new ThreadPoolExecutor(MAX_MERGE_SEND_THREAD,
-            MAX_MERGE_SEND_THREAD,
-            KEEP_ALIVE_TIME, TimeUnit.MILLISECONDS,
-            new LinkedBlockingQueue<>(),
-            new NamedThreadFactory(getThreadPrefix(), MAX_MERGE_SEND_THREAD));
-        mergeSendExecutorService.submit(new MergedSendRunnable());
+        if (NettyClientConfig.isEnableClientBatchSendRequest()) {
+            mergeSendExecutorService = new ThreadPoolExecutor(MAX_MERGE_SEND_THREAD,
```
Comment (l81893521) [inline core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemotingClient.java]: I m wondering why close the mergeSendExecutorService? If disable "isEnableClientBatchSendRequest", the baseketMap always empty.

Comment (objcoding) [reply core/src/main/java/io/seata/core/rpc/netty/AbstractRpcRemotingClient.java]: > I m wondering why close the mergeSendExecutorService? If disable "isEnableClientBatchSendRequest", the baseketMap always empty.

If it is a single send, you do not need mergeSendExecutorService

**Comment by l81893521** [comment]
I think should add the config to file.conf.example

**Pull request merged**